### PR TITLE
Rename pallet-simple-nft -> pallet-utxo-nft

### PR DESCRIPTION
**Pull request by jonmattgray**
`pallet-simple-nft` -> `pallet-utxo-nft`
`SimpleNFT` -> `UtxoNFT`

### CODE REVIEW

**Comment by jonmattgray** [comment]
Won't merge until rename PRs in `dscp-api` and `dscp-matchmaker-api` are ready to go

**Comment by jonmattgray** [comment]
Accompanying PRs which should pass once this PR merges and releases
https://github.com/digicatapult/dscp-api/pull/90
https://github.com/digicatapult/dscp-matchmaker-api/pull/62

**Review by mattdean-digicatapult — APPROVED** [review]

**Pull request merged**
